src/test/util/mining.cpp

```diff
@@ -22,19 +25,62 @@ CTxIn generatetoaddress(const NodeContext& node, const std::string& address)
     return MineBlock(node, coinbase_script);
 }
 
-CTxIn MineBlock(const NodeContext& node, const CScript& coinbase_scriptPubKey)
+void ReGenerateCommitments(CBlock& block)
+{
+    CMutableTransaction tx{*block.vtx.at(0)};
+    tx.vout.erase(tx.vout.begin() + GetWitnessCommitmentIndex(block));
```

Comment:
use .front()

Reply:
why?

.at() asserts that the element exists, which I like

Reply:
> why?
> 
> .at() asserts that the element exists, which I like

what do you mean by this?

Reply:
.at() will throw an exception when the vector is empty; front() will just cause UB.

Reply:
> .at() will throw an exception when the vector is empty; front() will just cause UB.

huh, did not know that. thanks for clarifying